### PR TITLE
CLOUDP-114869: Add decrypt command to OM and CM

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_[ CLOUDP-114869](https://jira.mongodb.org/browse/CLOUDP-114869)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/flag/flags.go [hunk 1]
```diff
@@ -271,4 +271,7 @@ const (
 	CurrentIP                                = "currentIp"                       // CurrentIP flag
 	Gov                                      = "gov"                             // Gov flag
 	Version                                  = "version"                         // Version flag
+	LocalKeyFile                             = "localKeyFile"                    // LocalKeyFile flag
+	KmipServerCAFile                         = "kimipServerCAFile"               // KmipServerCAFile flag
+	KmipClientCertificateFile                = "kmipClientCertificateFile"       // KmipClientCertificateFile flag
```
Comment (gssbzn) [inline internal/flag/flags.go:276]:
```suggestion
	KMIPServerCAFile                         = "kimipServerCAFile"               // KMIPServerCAFile flag
	KMIPClientCertificateFile                = "kmipClientCertificateFile"       // KMIPClientCertificateFile flag
```

File: internal/usage/usage.go [hunk 2]
```diff
@@ -309,4 +309,8 @@ const (
 	CurrentIP                                = "Flag that indicates whether to use the IP Address from the host that is currently executing the command. Only applicable for type ipAddress entries. To learn more, see: https://docs.mongodb.com/mongocli/master/command/mongocli-atlas-accessLists-create/."
 	Gov                                      = "Create a default profile for atlas for gov"
 	EncryptedLogFile                         = "Path to the file that contains encrypted audit logs."
+	OutputLogFile                            = "Path to the file where the decrypted audit log will be written. If not specified the default would be to write to stdout."
+	LocalKeyFile                             = "Path to the file that contains the Key Encryption Key (used to encrypt the Log Encryption Key)."
+	KmipServerCAFile                         = "Path to the CA file used to connect to the KMIP speaking server."
+	KmipClientCertificateFile                = "Path to the Client Cert file used to connect to the KMIP speaking server."
```
Comment (gssbzn) [inline internal/usage/usage.go:315]:
```suggestion
	KMIPServerCAFile                         = "Path to the CA file used to connect to the KMIP speaking server."
	KMIPClientCertificateFile                = "Path to the Client Cert file used to connect to the KMIP speaking server."
```

File: internal/cli/opsmanager/logs/decrypt.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	}
+
+	cmd.Flags().StringVarP(&opts.inFileName, flag.File, flag.FileShort, "", usage.EncryptedLogFile)
+	cmd.Flags().StringVarP(&opts.outFileName, flag.Out, "", "", usage.OutputLogFile)
+
+	cmd.Flags().StringVarP(&opts.localKeyFileName, flag.LocalKeyFile, "", "", usage.LocalKeyFile)
+	cmd.Flags().StringVarP(&opts.kmipServerCAFileName, flag.KmipServerCAFile, "", "", usage.KmipServerCAFile)
+	cmd.Flags().StringVarP(&opts.kmipClientCertificateFileName, flag.KmipClientCertificateFile, "", "", usage.KmipClientCertificateFile)
+
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
```
Comment (gssbzn) [inline internal/cli/opsmanager/logs/decrypt.go:76]: [q] wouldn't out be required as well?

File: internal/cli/opsmanager/logs/decrypt.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		return nil
+	}
+}
+
+func (opts *DecryptOpts) initFiles() func() error {
+	// Validate the provided files can be open
+	return func() error {
+		return nil
+	}
+}
```
Comment (gssbzn) [inline internal/cli/opsmanager/logs/decrypt.go:46]: FYI for the output file you may want to reuse https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/cli/downloader_opts.go which may need a rename

**Review by gssbzn — APPROVED** [review]
LGTM

File: internal/cli/opsmanager/logs/decrypt.go (hunk 4 above)
Comment (blva) [reply internal/cli/opsmanager/logs/decrypt.go:46]: Cool! Thanks I'll do that in the next PRs, for now want to keep as simple as possible if that's OK.

**Pull request merged**